### PR TITLE
[threadpool-ms] MonoError-ize

### PR DESCRIPTION
**Pull request by lambdageek**
1. Convert icalls that throw `NotImplementedException` to do it via mono_error_set_pending_exception
2. Add MonoError outargs to `mono_threadpool_ms_enqueue_work_item` and `mono_threadpool_ms_begin_invoke`

### CODE REVIEW

File: mono/metadata/threadpool-ms.c [hunk 1]
```diff
@@ -1583,7 +1583,9 @@ void
 ves_icall_System_Threading_ThreadPool_ReportThreadStatus (MonoBoolean is_working)
 {
 	// TODO
-	mono_raise_exception (mono_get_exception_not_implemented (NULL));
+	MonoError error;
+	mono_error_set_generic_error (&error, "System", "NotImplementedException", "");
+	mono_error_set_pending_exception (&error);
```
Comment (vargaz) [inline mono/metadata/threadpool-ms.c]: It would be nice to add helper functions to this instead of duplicating these strings.

**Comment by lambdageek** [comment]
@vargaz Yea, not a bad idea; I added some helpers in #2790 and updated this PR to use them.

File: mono/metadata/console-unix.c [hunk 2]
```diff
@@ -254,7 +254,8 @@ do_console_cancel_event (void)
 	method = mono_class_get_method_from_name (klass, "BeginInvoke", -1);
 	g_assert (method != NULL);
 
-	mono_threadpool_ms_begin_invoke (domain, (MonoObject*) load_value, method, NULL);
+	mono_threadpool_ms_begin_invoke (domain, (MonoObject*) load_value, method, NULL, &error);
+	mono_error_cleanup (&error);
```
Comment (kumpera) [inline mono/metadata/console-unix.c]: Swallowing the error.

Comment (lambdageek) [inline mono/metadata/console-unix.c]: What am I allowed to do on the finalizer thread? I can't raise here, can I? The best I can do is `g_warning (... , mono_error_get_message(&error))`, I think?

**Comment by lambdageek** [comment]
@monojenkins merge

File: mono/metadata/console-unix.c (hunk 2 above)
Comment (kumpera) [inline mono/metadata/console-unix.c]: Well, what's the expected behavior.
Maybe an exception/error on that case should abort the runtime.

**Pull request merged**